src/lib/pgp-key.c

```diff
@@ -776,3 +764,78 @@ pgp_pk_alg_capabilities(pgp_pubkey_alg_t alg)
         return PGP_KF_NONE;
     }
 }
+
+bool
+pgp_key_is_locked(const pgp_key_t *key)
+{
+    if (!pgp_is_key_secret(key)) {
+        RNP_LOG("key is not a secret key");
+        return false;
+    }
+    return key->key.seckey.encrypted;
+}
+
+bool
+pgp_key_unlock(pgp_key_t *key, const pgp_passphrase_provider_t *provider)
+{
+    pgp_seckey_t *decrypted_seckey = NULL;
+    bool          ok = false;
```

Comment:
``ok`` doesn't seem really useful here. ``decrypted_seckey`` will be set in case it was correctly decrypted and won't be set in case key isn't locked.